### PR TITLE
feat(navigation-logo): add heading level

### PR DESCRIPTION
**Pull request by josercarcamo**
**Related Issue:** #7075 

## Summary
Added the "headingLevel" property to navigation-logo, similar to Panel.

### CODE REVIEW

File: packages/calcite-components/src/components/navigation-logo/navigation-logo.tsx [hunk 1]
```diff
@@ -107,34 +113,45 @@ export class CalciteNavigationLogo implements LoadableComponent {
     return <calcite-icon class={CSS.icon} flipRtl={this.iconFlipRtl} icon={this.icon} scale="l" />;
   }
 
+  renderHeaderContent(): VNode {
+    const { heading, headingLevel, description } = this;
+    const headingNode = heading ? (
+      <Heading class={CSS.heading} level={headingLevel}>
+        <span
+          aria-label={this.heading}
```
Comment (driskull) [inline packages/calcite-components/src/components/navigation-logo/navigation-logo.tsx:121]: I don't think we need an aria-label here. The heading element should already provide enough information.

File: packages/calcite-components/src/components/navigation-logo/navigation-logo.tsx [hunk 2]
```diff
@@ -107,34 +113,45 @@ export class CalciteNavigationLogo implements LoadableComponent {
...
+          }}
+          key={CSS.heading}
+        >
+          {heading}
+        </span>
+      </Heading>
+    ) : null;
+
+    const descriptionNode = description ? (
+      <span aria-label={this.description} class={CSS.description} key={CSS.description}>
```
Comment (driskull) [inline packages/calcite-components/src/components/navigation-logo/navigation-logo.tsx:134]: Whats the need for the aria-label here as well?

**Review by driskull — CHANGES_REQUESTED** [review]
Looking good @josercarcamo 👍

File: packages/calcite-components/src/components/navigation-logo/navigation-logo.tsx (hunk 2 above)
Comment (josercarcamo) [reply packages/calcite-components/src/components/navigation-logo/navigation-logo.tsx:134]: I was just trying to keep the same sematics as what was in the original code, but I've removed it.

File: packages/calcite-components/src/components/navigation-logo/navigation-logo.tsx [hunk 3]
```diff
@@ -59,6 +60,11 @@ export class CalciteNavigationLogo implements LoadableComponent {
   /** Specifies the `src` to an image. */
   @Prop() thumbnail: string;
 
+  /**
+   * Specifies the number at which section headings should start.
```
Comment (geospatialem) [inline packages/calcite-components/src/components/navigation-logo/navigation-logo.tsx:64]: @josercarcamo Very timely! @DitwanP made similar changes across the components earlier today. CC'ing him for editorial review on the `headingLevel` doc update.

**Comment by anveshmekala** [comment]
Nitpick: PR title should be `feat` since this is an enhancement.

File: packages/calcite-components/src/components/navigation-logo/navigation-logo.tsx [hunk 4]
```diff
@@ -107,34 +113,43 @@ export class CalciteNavigationLogo implements LoadableComponent {
...
+    ) : null;
+
+    const descriptionNode = description ? (
+      <span class={CSS.description} key={CSS.description}>
+        {description}
+      </span>
+    ) : null;
+
+    return headingNode || descriptionNode ? (
+      <div class={CSS.container}>
```
Comment (driskull) [inline packages/calcite-components/src/components/navigation-logo/navigation-logo.tsx:138]: nitpick: should we give this one a key for consistency? `<div class={CSS.container} key={CSS.container}>`. up to you :)

**Review by driskull — CHANGES_REQUESTED** [review]
Can you add an test within the e2e test that tests the headingLevel? This would be using the `reflects` common test helper. You can see an example in the `block.e2e.ts`. 

```
 describe("reflects", () => {
    reflects("calcite-block", [
      {
        propertyName: "collapsible",
        value: true,
      },
      {
        propertyName: "headingLevel",
        value: 2,
      },
```

Also a "defaults" test would be good too.

```
describe("defaults", () => {
    defaults("calcite-block", [
      {
        propertyName: "collapsible",
        defaultValue: false,
      },
      {
        propertyName: "headingLevel",
        defaultValue: undefined,
      },
```

Can you also add a storybook story test in the `navigation-logo.stories.ts` that tests with a headingLevel applied? This way we can make sure different heading levels don't render differently.

Something like:

```
export const headingLevel_TestOnly = (): string => html`
 <calcite-navigation-logo
      heading="ArcGIS Online"
      heading-level="2"
      description="City of AcmeCo"
      thumbnail="${placeholderImage({ width: 50, height: 50 })}"
    />
`;
```

**Review by driskull — APPROVED** [review]
Looks good to me! 👍

**Pull request merged**